great_expectations/core/usage_statistics/execution_environment.py

```diff
@@ -0,0 +1,125 @@
+    version: Optional[version.Version]
+
+
+class GEExecutionEnvironment:
+    """The list of GE dependencies with version and install information.
+
+    This does not return any dependencies that are not specified directly in
+    either requirements.txt or any requirements-dev*.txt files.
+
+    Attributes: None
+    """
+
+    def __init__(self):
+        self._ge_dependencies: GEDependencies = GEDependencies()
+        self._all_installed_packages = None
+        self._get_all_installed_packages()
+
+        self._dependencies = []
+        self._build_required_dependencies()
+        self._build_dev_dependencies()
+
+    @property
+    def dependencies(self) -> List[PackageInfo]:
+        """The list of GE dependencies with version and install information.
+
+        This does not return any dependencies that are not specified directly in
+        either requirements.txt or any requirements-dev*.txt files.
+
+        Attributes: None
+        """
+        return self._dependencies
+
+    def _build_required_dependencies(self) -> None:
+        dependency_list: List[PackageInfo] = self._build_dependencies_info(
+            self._ge_dependencies.get_required_dependency_names(),
+            install_environment=InstallEnvironment.REQUIRED,
+        )
+        self._dependencies.extend(dependency_list)
+
+    def _build_dev_dependencies(self) -> None:
+        dependency_list: List[PackageInfo] = self._build_dependencies_info(
+            self._ge_dependencies.get_dev_dependency_names(),
+            install_environment=InstallEnvironment.DEV,
+        )
+        self._dependencies.extend(dependency_list)
+
+    def _get_all_installed_packages(self) -> List[str]:
+        """Get all installed packages. Used to check if dependency is installed or not.
+
+        Returns:
+            list of string names of packages that are installed
+        """
+        if self._all_installed_packages is None:
+            # Only retrieve once
+            self._all_installed_packages = [
+                item.metadata.get("Name") for item in metadata.distributions()
+            ]
+        return self._all_installed_packages
+
+    def _build_dependencies_info(
+        self, dependency_names: List[str], install_environment: InstallEnvironment
+    ) -> List[PackageInfo]:
+        """Build list of info about dependencies including version and install status.
+
+        Args:
+            dependency_names: Retrieve info on these dependencies.
+            install_environment: Environment these dependencies are installed in.
+
+        Returns:
+            List of PackageInfo objects for each dependency including version,
+            install status, install environment.
+        """
+        dependencies: List[PackageInfo] = []
+        for dependency_name in dependency_names:
+
+            package_version: Optional[version.Version]
+            installed: bool
+            if dependency_name in self._get_all_installed_packages():
```

Comment:
will we ever run this method without having first run `self._get_all_installed_packages` in the constructor? If not, could we just access `self._all_installed_packages`?

Reply:
We wouldn't but I was just hoping to have a single access point for the installed packages which enforces the singleton-like behavior (only retrieve once otherwise return cached value)

Reply:
I'm totally cool with this, thanks for the explanation! Feel free to resolve.